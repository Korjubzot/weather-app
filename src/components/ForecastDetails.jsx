import React from "react";

function ForecastDetails(props) {
  const { date, temperature, humidity, wind } = props;
  const { min, max } = temperature;
  const { speed: windspeed, direction } = wind;

  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details">
      <div
        className="forecast-details__date"
        data-testid="forecast-details__date"
      >
        {formattedDate}
      </div>
      <div
        className="forecast-details__minTemp"
        data-testid="forecast-details__minTemp"
      >
        {min}&deg;C
      </div>
      <div
        className="forecast-details__maxTemp"
        data-testid="forecast-details__maxTemp"
      >
        {max}&deg;C
      </div>
      <div
        className="forecast-details__humidity"
        data-testid="forecast-details__humidity"
      >
        {humidity}%
      </div>
      <div
        className="forecast-details__windspeed"
        data-testid="forecast-details__windspeed"
      >
        {windspeed} km/h
      </div>
      <div
        className="forecast-details__direction"
        data-testid="forecast-details__direction"
      >
        {direction}
      </div>
    </div>
  );
}

export default ForecastDetails;
