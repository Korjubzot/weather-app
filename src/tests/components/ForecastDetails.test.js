/* eslint-disable react/jsx-filename-extension */
import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("Forecast Details", () => {
  const validProps = {
    date: "Thu Jan 01 1970",
    temperature: {
      min: 11,
      max: 24,
    },
    humidity: 20,
    wind: {
      speed: 10,
      direction: "n",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastDetails
        date={validProps.date}
        temperature={validProps.temperature}
        humidity={validProps.humidity}
        wind={validProps.wind}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
