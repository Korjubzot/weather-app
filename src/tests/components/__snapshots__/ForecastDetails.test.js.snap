// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Details renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
      data-testid="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__minTemp"
      data-testid="forecast-details__minTemp"
    >
      11°C
    </div>
    <div
      class="forecast-details__maxTemp"
      data-testid="forecast-details__maxTemp"
    >
      24°C
    </div>
    <div
      class="forecast-details__humidity"
      data-testid="forecast-details__humidity"
    >
      20%
    </div>
    <div
      class="forecast-details__windspeed"
      data-testid="forecast-details__windspeed"
    >
      10 km/h
    </div>
    <div
      class="forecast-details__direction"
      data-testid="forecast-details__direction"
    >
      n
    </div>
  </div>
</DocumentFragment>
`;
